src/iter/unzip.rs

```diff
@@ -386,3 +395,57 @@ impl<A, B, RA, RB> Reducer<(A, B)> for UnzipReducer<RA, RB>
         (self.left.reduce(left.0, right.0), self.right.reduce(left.1, right.1))
     }
 }
+
+
+impl<A, B, FromA, FromB> ParallelExtend<(A, B)> for (FromA, FromB)
+where
+    A: Send,
+    B: Send,
+    FromA: Send + ParallelExtend<A>,
+    FromB: Send + ParallelExtend<B>,
+{
+    fn par_extend<I>(&mut self, pi: I)
+    where
+        I: IntoParallelIterator<Item = (A, B)>,
+    {
+        execute_into(&mut self.0, &mut self.1, pi.into_par_iter(), Unzip);
+    }
+}
+
+impl<L, R, A, B> ParallelExtend<Either<L, R>> for (A, B)
```

Comment:
I think coherence rules would make this impossible for `Extend` since tuples aren't fundamental -- the `either` crate can't implement the trait for a generic tuple type.  So this is a unique possibility for `ParallelExtend`!

Reply:
Hmm, Rust 1.41's "re-rebalance coherence" would allow this now... :thinking: